### DISCUSSION TITLE
Cant Upload Big Files

### DISCUSSION THREAD

**Discussion by GizzyUwU** [git-lfs/git-lfs#4844] [Q&A]
C:\Users\offic\Desktop\Zips>git lfs track "brave.zip"
"brave.zip" already supported

C:\Users\offic\Desktop\Zips>git commit -m "Brave Test"
On branch master
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .gitattributes
        git.html.txt

nothing added to commit but untracked files present (use "git add" to track)

C:\Users\offic\Desktop\Zips>git push origin master
Enumerating objects: 3, done.
Counting objects: 100% (3/3), done.
Delta compression using up to 4 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 120.06 MiB | 925.00 KiB/s, done.
Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
remote: error: Trace: d96fcede7b71e4fa022ff7e19b4aa98594f1fd4c9ac4e075086b341cbb4bd1b2
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File Brave.zip is 120.22 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/MCRaspRBX/WinbookBypass/
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/MCRaspRBX/WinbookBypass/'

**Comment by bk2204** [comment]
Hey,

There are two issues you have here that are preventing this from working.

First, when you run `git lfs track`, you have to add and commit the `.gitattributes` file, which tells Git that a file is an LFS file.  However, in your example, you haven't added it.

Second, `git lfs track` doesn't modify existing history, so if you've already added a large file in the history, it won't prevent the push from being rejected.  If you want to modify existing history, then you'd want to run `git lfs migrate import --everything --include="brave.zip"`, which will rewrite the entire history to make it track `brave.zip`.

A final note: if your file is actually called `Brave.zip` and not `brave.zip`, then you should use the former, or your repository may not work correctly.  Those may be the same on Windows, but they are not on other systems or under the Windows Subsystem for Linux, and since Git is case sensitive, you can potentially be in for a world of pain if you don't get the case right.

**Reply by ahmad-alismail** [reply]
Hello, I have a similar error.  Could you help me please?
````
ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git lfs track "model.pth.tar"
Tracking "model.pth.tar"

ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git add .gitattributes

ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git add model.pth.tar

ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git status
On branch master
Your branch is ahead of 'origin/master' by 1 commit.
  (use "git push" to publish your local commits)

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   .gitattributes
        modified:   model.pth.tar


ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git commit -m "upload the model"
[master 33333cc] upload the model
 2 files changed, 1 insertion(+)
 rewrite model.pth.tar (99%)


ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git push origin master
Uploading LFS objects: 100% (1/1), 149 MB | 1.3 MB/s, done.
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Delta compression using up to 8 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 81.51 MiB | 1.48 MiB/s, done.
Total 8 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), done.
remote: error: Trace: f94efc6d7fb21236c372de050feabf094ecc437130dd467a3e02568e1a88a868
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File model.pth.tar is 142.31 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/afi1289/model_deployment.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/afi1289/model_deployment.git'
````

**Reply by bk2204** [reply]
Hey,

The same thing applies here, except that you'll run `git lfs migrate import --everything --include="model.pth.tar"`.

**Reply by ahmad-alismail** [reply]
Thank you for your reply! Sorry maybe I'm beginner in git to understand this point. Where I should add ``git lfs migrate import --everything --include="model.pth.tar"`` ? Should I replace ``git lfs track "model.pth.tar"``? I've been trying 7 hours ago to upload this file only...

**Reply by bk2204** [reply]
That command will perform the tracking for you.

**Reply by ahmad-alismail** [reply]
Unfortunately, It doesn't work for me... The steps were as follows:

````
ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git lfs install
Updated Git hooks.
Git LFS initialized.

ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git lfs migrate import --everything --include="model.pth.tar"
migrate: override changes in your working copy?  All uncommitted changes will be lost! [y/N] y
migrate: changes in your working copy will be overridden ...
migrate: Sorting commits: ..., done.
migrate: Rewriting commits: 100% (1/1), done.
  master        56a3002e9c66e58090ed589678a30784289f8a5b -> 570ae1e3e4db6d515b789492fe52d5cc4a5be273
migrate: Updating refs: ..., done.
migrate: checkout: ..., done.

ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git add .gitattributes

ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git add model.pth.tar

ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git commit -m "Add model file"
[master 1df28d8] Add model file
 1 file changed, 3 insertions(+)
 create mode 100644 model.pth.tar

ahmad.alismail@W0036 MINGW64 ~/Documents/My-Github/model_deployment/model_deployment (master)
$ git push origin master
To https://github.com/afi1289/model_deployment.git
 ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs to 'https://github.com/afi1289/model_deployment.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

````

**Reply by bk2204** [reply]
This rewrites history, so you need to do `git push --force origin master`.
